File: AspNetMvcUnleashed/0672329980_source/Chapter06/CS/MvcApplication1/MvcApplication1/Controllers/ProductController.cs
using System.Collections.Generic;
using System.Linq;
using System.Web.Mvc;
using MvcApplication1.Models;
using Paging;

namespace MvcApplication1.Controllers
{
    public class ProductController : Controller
    {
        private ToyStoreDBEntities _entities = new ToyStoreDBEntities(); 

        public ActionResult Index()
        {
            return View(_entities.ProductSet.ToList());
        }


        public ActionResult List()
        {
            ViewData["products"] = _entities.ProductSet.ToList(); 
            return View();
        }



        public ActionResult SortProducts(string sort)
        {
            IEnumerable<Product> products;
            sort = sort ?? string.Empty;
            switch (sort.ToLower())
            {
                case "name":
                    products = from p in _entities.ProductSet 
                               orderby p.Name select p;
                    break;
                case "price":
                    products = from p in _entities.ProductSet
                               orderby p.Price
                               select p;
                    break;
                default:
                    products = from p in _entities.ProductSet
                               orderby p.Id
                               select p;
                    break;
            }

            return View(products);
        }


        public ActionResult PagedProducts(int? page)
        {
            var products = _entities.ProductSet
                .OrderBy(p => p.Id).ToPagedList(page, 2);
            
            return View(products);
        }


        public ActionResult PagedSortedProducts(string sort, int? page)
        {
            IQueryable<Product> products;
            sort = sort ?? string.Empty;
            switch (sort.ToLower())
            {
                case "name":
                    products = from p in _entities.ProductSet
                               orderby p.Name
                               select p;
                    break;
                case "price":
                    products = from p in _entities.ProductSet
                               orderby p.Price
                               select p;
                    break;
                default:
                    products = from p in _entities.ProductSet
                               orderby p.Id
                               select p;
                    break;
            }

            ViewData.Model = products.ToPagedList(page, 2, sort);
            return View();
        }






    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter06/VB/MvcApplication1/MvcApplication1/Controllers/ProductController.vb
Public Class ProductController
    Inherits System.Web.Mvc.Controller

    Private _entities As New ToyStoreDBEntities()

    Function Index() As ActionResult
        Return View(_entities.ProductSet.ToList())
    End Function

    
    Function List() As ActionResult
        ViewData("products") = _entities.ProductSet.ToList()
        Return View()
    End Function

    Function SortProducts(ByVal sort As String) As ActionResult
	    Dim products As IEnumerable(Of Product)
	    sort = If((sort <> Nothing), sort, String.Empty)
	    Select Case sort.ToLower()
		    Case "name"
			    products = From p In _entities.ProductSet _
			               Order By p.Name _
			               Select p
		    Case "price"
			    products = From p In _entities.ProductSet _
			               Order By p.Price _
			               Select p
		    Case Else
			    products = From p In _entities.ProductSet _
			               Order By p.Id _
			               Select p
	    End Select

	    Return View(products)
    End Function

    Function PagedProducts(ByVal page As Integer?) As ActionResult
        Dim products = _entities.ProductSet _
                .OrderBy(Function(p) p.Id) _
                .ToPagedList(page, 2)

	    Return View(products)
    End Function


    Function PagedSortedProducts(ByVal sort As String, ByVal page As Integer?) As ActionResult
	    Dim products As IQueryable(Of Product)
	    sort = If((sort <> Nothing), sort, String.Empty)
	    Select Case sort.ToLower()
		    Case "name"
			    products = From p In _entities.ProductSet _
			               Order By p.Name _
			               Select p
		    Case "price"
			    products = From p In _entities.ProductSet _
			               Order By p.Price _
			               Select p
		    Case Else
			    products = From p In _entities.ProductSet _
			               Order By p.Id _
			               Select p
	    End Select

	    ViewData.Model = products.ToPagedList(page, 2, sort)
	    Return View()
    End Function



End Class
